cypress/support/cypress-commands/lifecycle.js

```diff
@@ -198,13 +198,18 @@ Cypress.Commands.add('validateLifecycleHistoryAndEvents', (state, appId) => {
       const lifecycleEventRequirementId = scenarioRequirement.find((req) =>
         req.hasOwnProperty('event')
       );
+      const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
+      const appHistoryCount = appHistory.length - appHistoryPrevious.length;
 
       // Lifecycle event validation
-      if (lifecycleEventRequirementId && lifecycleEventRequirementId.event) {
-        const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
-        const appHistoryCount = appHistory.length - appHistoryPrevious.length;
+      if (condition == CONSTANTS.STAY) {
+        UTILS.assertWithRequirementLogs(
+          CONSTANTS.LIFECYCLE_NOTIFICATION_NOT_GENERATED,
+          appHistoryCount < 1,
+          true
+        );
+      } else if (lifecycleEventRequirementId && lifecycleEventRequirementId.event) {
```

Comment:
why there is line 214 which uses IS_SAME_APP_TRANSITION

Reply:
isnt that the requirement mentioned in the ticket / to remove the variable

Reply:
updated as per requirement